### ISSUE TITLE
SeekableCallbackMedia stops reading buffer

### ISSUE THREAD

**Issue by roiniti** [caprica/vlcj#1193]
Hi,
I'm building my own SeekableCallbackMedia for playing custom web streams but in some point the video stops playing, debugging I saw that its not problem of the web stream, the problem resides in that onRead is called until x point reading the buffer but then its not called anymore causing the video stop like it has reached the end earlier than normal, i don't know why this happens.
I also tested with using RandomAccessFileMedia instead, but i have the same problem.
I'm doing this to play the video
`capricaMediaPlayer.media().prepare(new RandomAccessFileMedia(new File("path to mp4 file")),options);`
and later i do
`capricaMediaPlayer.controls().play();`
where caprocaMediaPlayer is an EmbeddedMediaPlayer created with the MediaPlayerFactory

**Comment by caprica** [comment]
This is difficult to troubleshoot, because it is actually VLC that calls back into the vlcj code to read the data.  It's not like vlcj is in control of when read gets called.

So it would appear that VLC has stopped asking the Java code for more data.

Without a repeatable test case it is going to be very difficult to look at this, and even then it may be a problem in VLC, not vlcj.

At the least you should run VisualVM or something similar and see if any threads are stuck.

**Comment by caprica** [comment]
If you play your stream in VLC and seek around, is it ok?

**Comment by roiniti** [comment]
I didn't tried in VLC, i need to send specific headers to access the stream but downloading the stream and then playing it in vlcj with .media().play(AbsolutePath) works, using VisualVM i see no much weird things, when i start the player i see three media-player-events threads running that runs until the video reaches the end (not the end of the stream, the end of the stream that it reads, that is 21910660 bytes of 83920527 total, the number of bytes read changes every time), i only notice that its creating 0ms of life threads when i use my implementation of SeekableCallbackMedia

**Comment by caprica** [comment]
Those Thread objects (not actual threads) are just an artefact of how JNA works.

**Comment by caprica** [comment]
Can you post a thread dump at the point it stops reading?

**Comment by roiniti** [comment]
I don't know if i did it correctly but here it is https://pastebin.com/wVHxEDg7 , I started the player with RandomAccessFileMedia of a mp4, 25 minutes, and then seeked arround the half of the video

**Comment by caprica** [comment]
You did it correctly, but you could've just posted it here.

**Comment by caprica** [comment]
I dunno, I don't see any vlcj classes in that dump. It's hard to say.

**Comment by roiniti** [comment]
Sorry i think its because the video ended playing because it reached the end, i paused the video waiting the buffer to get filled, this is before it reaches the end of the buffer, the video is still playing at this moment but the buffer has ended its reading i think 
`
2023-05-14 22:52:54
Full thread dump Java HotSpot(TM) 64-Bit Server VM (11.0.11+9-LTS-194 mixed mode):

Threads class SMR info:
_java_thread_list=0x00000156fbf3ddd0, length=30, elements={
0x00000156dea43800, 0x00000156f94a5000, 0x00000156f94ae000, 0x00000156f94c2000,
0x00000156f94c4000, 0x00000156f94c5800, 0x00000156f94d1800, 0x00000156f94d6800,
0x00000156f9fc1000, 0x00000156f9fca000, 0x00000156faa6d000, 0x00000156fa8ce800,
0x00000156fa68a000, 0x00000156fbfaa800, 0x00000156fc9be000, 0x00000156fd957000,
0x00000156fdf5d800, 0x000001568721e000, 0x00000156fdfc9000, 0x00000156fdfc7800,
0x00000156fdfc5000, 0x00000156fdfc6000, 0x00000156fdfc8800, 0x00000156fbd2a000,
0x00000156fbd27800, 0x00000156fbd2b800, 0x00000156fbd2c800, 0x00000156fbd28800,
0x00000156fbd2d800, 0x00000156fbd2f000
}

"app thread" #1 prio=5 os_prio=0 cpu=140.63ms elapsed=238.32s tid=0x00000156dea43800 nid=0x90fc in Object.wait()  [0x000000364f8fe000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <0x00000000f8328180> (a java.lang.Object)
        at java.lang.Object.wait(java.base@11.0.11/Object.java:328)
        at net.roiniti.jams.desktop.DesktopLauncher.main(DesktopLauncher.java:273)
        - waiting to re-lock in wait() <0x00000000f8328180> (a java.lang.Object)

   Locked ownable synchronizers:
        - None

"Reference Handler" #2 daemon prio=10 os_prio=2 cpu=0.00ms elapsed=238.29s tid=0x00000156f94a5000 nid=0x63b4 waiting on condition  [0x000000364fffe000]
   java.lang.Thread.State: RUNNABLE
        at java.lang.ref.Reference.waitForReferencePendingList(java.base@11.0.11/Native Method)
        at java.lang.ref.Reference.processPendingReferences(java.base@11.0.11/Reference.java:241)
        at java.lang.ref.Reference$ReferenceHandler.run(java.base@11.0.11/Reference.java:213)

   Locked ownable synchronizers:
        - None

"Finalizer" #3 daemon prio=8 os_prio=1 cpu=0.00ms elapsed=238.29s tid=0x00000156f94ae000 nid=0x91cc in Object.wait()  [0x00000036500fe000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <no object reference available>
        at java.lang.ref.ReferenceQueue.remove(java.base@11.0.11/ReferenceQueue.java:155)
        - waiting to re-lock in wait() <0x00000000f832a208> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(java.base@11.0.11/ReferenceQueue.java:176)
        at java.lang.ref.Finalizer$FinalizerThread.run(java.base@11.0.11/Finalizer.java:170)

   Locked ownable synchronizers:
        - None

"Signal Dispatcher" #4 daemon prio=9 os_prio=2 cpu=0.00ms elapsed=238.27s tid=0x00000156f94c2000 nid=0x9364 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"Attach Listener" #5 daemon prio=5 os_prio=2 cpu=15.63ms elapsed=238.27s tid=0x00000156f94c4000 nid=0x78d8 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"C2 CompilerThread0" #6 daemon prio=9 os_prio=2 cpu=5468.75ms elapsed=238.27s tid=0x00000156f94c5800 nid=0x95d8 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

   Locked ownable synchronizers:
        - None

"C1 CompilerThread0" #9 daemon prio=9 os_prio=2 cpu=265.63ms elapsed=238.27s tid=0x00000156f94d1800 nid=0x93e0 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE
   No compile task

   Locked ownable synchronizers:
        - None

"Sweeper thread" #10 daemon prio=9 os_prio=2 cpu=0.00ms elapsed=238.27s tid=0x00000156f94d6800 nid=0x7970 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"Service Thread" #11 daemon prio=9 os_prio=0 cpu=0.00ms elapsed=238.18s tid=0x00000156f9fc1000 nid=0x97c4 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"Common-Cleaner" #12 daemon prio=8 os_prio=1 cpu=0.00ms elapsed=238.18s tid=0x00000156f9fca000 nid=0x8cd0 in Object.wait()  [0x00000036508fe000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <no object reference available>
        at java.lang.ref.ReferenceQueue.remove(java.base@11.0.11/ReferenceQueue.java:155)
        - waiting to re-lock in wait() <0x00000000f832e6f0> (a java.lang.ref.ReferenceQueue$Lock)
        at jdk.internal.ref.CleanerImpl.run(java.base@11.0.11/CleanerImpl.java:148)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)
        at jdk.internal.misc.InnocuousThread.run(java.base@11.0.11/InnocuousThread.java:134)

   Locked ownable synchronizers:
        - None

"Java2D Disposer" #13 daemon prio=10 os_prio=2 cpu=0.00ms elapsed=237.77s tid=0x00000156faa6d000 nid=0x9180 in Object.wait()  [0x0000003650afe000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <0x00000000f8328ca0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(java.base@11.0.11/ReferenceQueue.java:155)
        - waiting to re-lock in wait() <0x00000000f8328ca0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(java.base@11.0.11/ReferenceQueue.java:176)
        at sun.java2d.Disposer.run(java.desktop@11.0.11/Disposer.java:144)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"AWT-Shutdown" #14 prio=5 os_prio=0 cpu=0.00ms elapsed=237.77s tid=0x00000156fa8ce800 nid=0x91b4 in Object.wait()  [0x0000003650bfe000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <0x00000000f83277c8> (a java.lang.Object)
        at java.lang.Object.wait(java.base@11.0.11/Object.java:328)
        at sun.awt.AWTAutoShutdown.run(java.desktop@11.0.11/AWTAutoShutdown.java:291)
        - waiting to re-lock in wait() <0x00000000f83277c8> (a java.lang.Object)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"AWT-Windows" #15 daemon prio=6 os_prio=0 cpu=15.63ms elapsed=237.76s tid=0x00000156fa68a000 nid=0x976c runnable  [0x0000003650cff000]
   java.lang.Thread.State: RUNNABLE
        at sun.awt.windows.WToolkit.eventLoop(java.desktop@11.0.11/Native Method)
        at sun.awt.windows.WToolkit.run(java.desktop@11.0.11/WToolkit.java:305)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"Discord Presence" #18 prio=5 os_prio=0 cpu=0.00ms elapsed=237.29s tid=0x00000156fbfaa800 nid=0x9390 waiting on condition  [0x0000003650eff000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
        at java.lang.Thread.sleep(java.base@11.0.11/Native Method)
        at net.roiniti.jams.DiscordJamsRCP$1.run(DiscordJamsRCP.java:45)

   Locked ownable synchronizers:
        - None

"AWT-EventQueue-0" #22 prio=6 os_prio=0 cpu=0.00ms elapsed=236.91s tid=0x00000156fc9be000 nid=0x9524 waiting on condition  [0x00000036514fe000]
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@11.0.11/Native Method)
        - parking to wait for  <0x00000000f8253698> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(java.base@11.0.11/LockSupport.java:194)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.11/AbstractQueuedSynchronizer.java:2081)
        at java.awt.EventQueue.getNextEvent(java.desktop@11.0.11/EventQueue.java:566)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(java.desktop@11.0.11/EventDispatchThread.java:190)
        at java.awt.EventDispatchThread.pumpEventsForFilter(java.desktop@11.0.11/EventDispatchThread.java:124)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(java.desktop@11.0.11/EventDispatchThread.java:113)
        at java.awt.EventDispatchThread.pumpEvents(java.desktop@11.0.11/EventDispatchThread.java:109)
        at java.awt.EventDispatchThread.pumpEvents(java.desktop@11.0.11/EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.run(java.desktop@11.0.11/EventDispatchThread.java:90)

   Locked ownable synchronizers:
        - None

"GDX-Thread" #20 prio=5 os_prio=0 cpu=2531.25ms elapsed=236.79s tid=0x00000156fd957000 nid=0x15dc runnable  [0x00000036521ff000]
   java.lang.Thread.State: RUNNABLE
        at org.lwjgl.system.JNI.invokePV(Native Method)
        at org.lwjgl.glfw.GLFW.glfwSwapBuffers(GLFW.java:4565)
        at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:404)
        at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:143)
        - locked <0x00000000f856ea08> (a com.badlogic.gdx.utils.Array)
        at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
        at net.roiniti.jams.desktop.AplicationThread$1.<init>(AplicationThread.java:44)
        at net.roiniti.jams.desktop.AplicationThread.run(AplicationThread.java:44)

   Locked ownable synchronizers:
        - None

"Mobile support" #29 prio=5 os_prio=0 cpu=0.00ms elapsed=235.79s tid=0x00000156fdf5d800 nid=0x7c8c runnable  [0x0000003652efe000]
   java.lang.Thread.State: RUNNABLE
        at java.net.PlainSocketImpl.waitForNewConnection(java.base@11.0.11/Native Method)
        at java.net.PlainSocketImpl.socketAccept(java.base@11.0.11/PlainSocketImpl.java:163)
        at java.net.AbstractPlainSocketImpl.accept(java.base@11.0.11/AbstractPlainSocketImpl.java:458)
        at java.net.ServerSocket.implAccept(java.base@11.0.11/ServerSocket.java:551)
        at java.net.ServerSocket.accept(java.base@11.0.11/ServerSocket.java:519)
        at com.badlogic.gdx.net.NetJavaServerSocketImpl.accept(NetJavaServerSocketImpl.java:80)
        at net.roiniti.jams.managers.MobileSupport.run(MobileSupport.java:70)
        at net.roiniti.jams.JAMS$2.run(JAMS.java:163)

   Locked ownable synchronizers:
        - None

"NotificateListener" #30 prio=5 os_prio=0 cpu=15.63ms elapsed=235.78s tid=0x000001568721e000 nid=0x484c waiting on condition  [0x0000003652ffe000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
        at java.lang.Thread.sleep(java.base@11.0.11/Native Method)
        at net.roiniti.jams.desktop.DesktopLauncher$2.run(DesktopLauncher.java:261)

   Locked ownable synchronizers:
        - None

"LWJGL3 Timer" #31 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=235.78s tid=0x00000156fdfc9000 nid=0x8674 waiting on condition  [0x00000036530ff000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
        at java.lang.Thread.sleep(java.base@11.0.11/Native Method)
        at com.badlogic.gdx.backends.lwjgl3.Sync$1.run(Sync.java:120)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"DownloadContrllerThread" #37 prio=5 os_prio=0 cpu=0.00ms elapsed=233.79s tid=0x00000156fdfc7800 nid=0x94c0 waiting on condition  [0x0000003652dfe000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
        at java.lang.Thread.sleep(java.base@11.0.11/Native Method)
        at net.roiniti.jams.managers.network.download.DownloadController$1.run(DownloadController.java:31)

   Locked ownable synchronizers:
        - None

"Drive stalker" #38 prio=5 os_prio=0 cpu=31.25ms elapsed=233.79s tid=0x00000156fdfc5000 nid=0x9680 runnable  [0x00000036532fe000]
   java.lang.Thread.State: RUNNABLE
        at java.net.SocketInputStream.socketRead0(java.base@11.0.11/Native Method)
        at java.net.SocketInputStream.socketRead(java.base@11.0.11/SocketInputStream.java:115)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:168)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:140)
        at sun.security.ssl.SSLSocketInputRecord.read(java.base@11.0.11/SSLSocketInputRecord.java:478)
        at sun.security.ssl.SSLSocketInputRecord.readHeader(java.base@11.0.11/SSLSocketInputRecord.java:472)
        at sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(java.base@11.0.11/SSLSocketInputRecord.java:70)
        at sun.security.ssl.SSLSocketImpl.readApplicationRecord(java.base@11.0.11/SSLSocketImpl.java:1318)
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(java.base@11.0.11/SSLSocketImpl.java:959)
        at java.io.BufferedInputStream.fill(java.base@11.0.11/BufferedInputStream.java:252)
        at java.io.BufferedInputStream.read1(java.base@11.0.11/BufferedInputStream.java:292)
        at java.io.BufferedInputStream.read(java.base@11.0.11/BufferedInputStream.java:351)
        - locked <0x00000000fd4100e8> (a java.io.BufferedInputStream)
        at sun.net.www.http.HttpClient.parseHTTPHeader(java.base@11.0.11/HttpClient.java:746)
        at sun.net.www.http.HttpClient.parseHTTP(java.base@11.0.11/HttpClient.java:689)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(java.base@11.0.11/HttpURLConnection.java:1615)
        - locked <0x00000000fe367818> (a sun.net.www.protocol.https.DelegateHttpsURLConnection)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(java.base@11.0.11/HttpURLConnection.java:1520)
        - locked <0x00000000fe367818> (a sun.net.www.protocol.https.DelegateHttpsURLConnection)
        at java.net.HttpURLConnection.getResponseCode(java.base@11.0.11/HttpURLConnection.java:527)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(java.base@11.0.11/HttpsURLConnectionImpl.java:334)
        at com.google.api.client.http.javanet.NetHttpResponse.<init>(NetHttpResponse.java:37)
        at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:105)
        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:981)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
        at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
        at net.roiniti.jams.drive.JamsDrive.getEsentialFile(JamsDrive.java:227)
        at net.roiniti.jams.drive.JamsDrive.access$1(JamsDrive.java:223)
        at net.roiniti.jams.drive.JamsDrive$3.run(JamsDrive.java:418)

   Locked ownable synchronizers:
        - <0x00000000fe368c60> (a java.util.concurrent.locks.ReentrantLock$NonfairSync)

"Timer" #103 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=229.30s tid=0x00000156fdfc6000 nid=0x928c in Object.wait()  [0x00000036520ff000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <no object reference available>
        at com.badlogic.gdx.utils.Timer$TimerThread.run(Timer.java:287)
        - waiting to re-lock in wait() <0x00000000f89bda58> (a java.lang.Object)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"RMI TCP Accept-0" #117 daemon prio=5 os_prio=0 cpu=218.75ms elapsed=224.35s tid=0x00000156fdfc8800 nid=0x8908 runnable  [0x00000036522ff000]
   java.lang.Thread.State: RUNNABLE
        at java.net.PlainSocketImpl.accept0(java.base@11.0.11/Native Method)
        at java.net.PlainSocketImpl.socketAccept(java.base@11.0.11/PlainSocketImpl.java:159)
        at java.net.AbstractPlainSocketImpl.accept(java.base@11.0.11/AbstractPlainSocketImpl.java:458)
        at java.net.ServerSocket.implAccept(java.base@11.0.11/ServerSocket.java:551)
        at java.net.ServerSocket.accept(java.base@11.0.11/ServerSocket.java:519)
        at sun.management.jmxremote.LocalRMIServerSocketFactory$1.accept(jdk.management.agent@11.0.11/LocalRMIServerSocketFactory.java:52)
        at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.executeAcceptLoop(java.rmi@11.0.11/TCPTransport.java:394)
        at sun.rmi.transport.tcp.TCPTransport$AcceptLoop.run(java.rmi@11.0.11/TCPTransport.java:366)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"RMI Scheduler(0)" #283 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=223.94s tid=0x00000156fbd2a000 nid=0x974c waiting on condition  [0x00000036509fe000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@11.0.11/Native Method)
        - parking to wait for  <0x00000000f8ce4dd0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.11/LockSupport.java:234)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.11/AbstractQueuedSynchronizer.java:2123)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.11/ScheduledThreadPoolExecutor.java:1182)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.11/ScheduledThreadPoolExecutor.java:899)
        at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.11/ThreadPoolExecutor.java:1054)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.11/ThreadPoolExecutor.java:1114)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.11/ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"JMX server connection timeout 284" #284 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=223.94s tid=0x00000156fbd27800 nid=0x930c in Object.wait()  [0x0000003653fff000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <no object reference available>
        at com.sun.jmx.remote.internal.ServerCommunicatorAdmin$Timeout.run(java.management@11.0.11/ServerCommunicatorAdmin.java:171)
        - waiting to re-lock in wait() <0x00000000f8eb0418> (a [I)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - None

"RMI TCP Connection(3)-169.254.244.72" #2486 daemon prio=5 os_prio=0 cpu=15.63ms elapsed=178.18s tid=0x00000156fbd2b800 nid=0x56ac runnable  [0x000000364f5fd000]
   java.lang.Thread.State: RUNNABLE
        at java.net.SocketInputStream.socketRead0(java.base@11.0.11/Native Method)
        at java.net.SocketInputStream.socketRead(java.base@11.0.11/SocketInputStream.java:115)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:168)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:140)
        at java.io.BufferedInputStream.fill(java.base@11.0.11/BufferedInputStream.java:252)
        at java.io.BufferedInputStream.read(java.base@11.0.11/BufferedInputStream.java:271)
        - locked <0x00000000f8cfcfc0> (a java.io.BufferedInputStream)
        at java.io.FilterInputStream.read(java.base@11.0.11/FilterInputStream.java:83)
        at sun.rmi.transport.tcp.TCPTransport.handleMessages(java.rmi@11.0.11/TCPTransport.java:544)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(java.rmi@11.0.11/TCPTransport.java:796)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(java.rmi@11.0.11/TCPTransport.java:677)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$$Lambda$244/0x000000010047fc40.run(java.rmi@11.0.11/Unknown Source)
        at java.security.AccessController.doPrivileged(java.base@11.0.11/Native Method)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(java.rmi@11.0.11/TCPTransport.java:676)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.11/ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.11/ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - <0x00000000f8ecd9f0> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"media-player-events" #2507 daemon prio=5 os_prio=0 cpu=46.88ms elapsed=170.81s tid=0x00000156fbd2c800 nid=0x8c10 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"media-player-events" #2688 daemon prio=5 os_prio=0 cpu=125.00ms elapsed=170.78s tid=0x00000156fbd28800 nid=0x71a0 runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"media-player-events" #2731 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=170.76s tid=0x00000156fbd2d800 nid=0x96fc runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

   Locked ownable synchronizers:
        - None

"RMI TCP Connection(4)-169.254.244.72" #10173 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=17.69s tid=0x00000156fbd2f000 nid=0x9370 runnable  [0x000000364f7fd000]
   java.lang.Thread.State: RUNNABLE
        at java.net.SocketInputStream.socketRead0(java.base@11.0.11/Native Method)
        at java.net.SocketInputStream.socketRead(java.base@11.0.11/SocketInputStream.java:115)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:168)
        at java.net.SocketInputStream.read(java.base@11.0.11/SocketInputStream.java:140)
        at java.io.BufferedInputStream.fill(java.base@11.0.11/BufferedInputStream.java:252)
        at java.io.BufferedInputStream.read(java.base@11.0.11/BufferedInputStream.java:271)
        - locked <0x00000000f8ee7258> (a java.io.BufferedInputStream)
        at java.io.FilterInputStream.read(java.base@11.0.11/FilterInputStream.java:83)
        at sun.rmi.transport.tcp.TCPTransport.handleMessages(java.rmi@11.0.11/TCPTransport.java:544)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(java.rmi@11.0.11/TCPTransport.java:796)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(java.rmi@11.0.11/TCPTransport.java:677)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler$$Lambda$244/0x000000010047fc40.run(java.rmi@11.0.11/Unknown Source)
        at java.security.AccessController.doPrivileged(java.base@11.0.11/Native Method)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(java.rmi@11.0.11/TCPTransport.java:676)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.11/ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.11/ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(java.base@11.0.11/Thread.java:834)

   Locked ownable synchronizers:
        - <0x00000000f8eeb878> (a java.util.concurrent.ThreadPoolExecutor$Worker)

"VM Thread" os_prio=2 cpu=78.13ms elapsed=238.29s tid=0x00000156f9482800 nid=0xf94 runnable  

"GC Thread#0" os_prio=2 cpu=31.25ms elapsed=238.31s tid=0x00000156dea59800 nid=0x56a0 runnable  

"GC Thread#1" os_prio=2 cpu=62.50ms elapsed=236.88s tid=0x00000156fc83c000 nid=0x5264 runnable  

"GC Thread#2" os_prio=2 cpu=62.50ms elapsed=236.87s tid=0x00000156fc79f000 nid=0x808c runnable  

"GC Thread#3" os_prio=2 cpu=109.38ms elapsed=236.87s tid=0x00000156fc79f800 nid=0x85f8 runnable  

"GC Thread#4" os_prio=2 cpu=46.88ms elapsed=236.87s tid=0x00000156fc7a7000 nid=0x8538 runnable  

"GC Thread#5" os_prio=2 cpu=93.75ms elapsed=236.87s tid=0x00000156fc73c000 nid=0x8388 runnable  

"GC Thread#6" os_prio=2 cpu=31.25ms elapsed=236.87s tid=0x00000156fc739800 nid=0x5e4c runnable  

"GC Thread#7" os_prio=2 cpu=125.00ms elapsed=236.87s tid=0x00000156fc739000 nid=0x7aa0 runnable  

"GC Thread#8" os_prio=2 cpu=0.00ms elapsed=236.87s tid=0x00000156fc73c800 nid=0x9700 runnable  

"GC Thread#9" os_prio=2 cpu=78.13ms elapsed=236.87s tid=0x00000156fc73d800 nid=0x8878 runnable  

"G1 Main Marker" os_prio=2 cpu=0.00ms elapsed=238.31s tid=0x00000156dea61800 nid=0x8fc8 runnable  

"G1 Conc#0" os_prio=2 cpu=0.00ms elapsed=238.31s tid=0x00000156dea64800 nid=0x9560 runnable  

"G1 Conc#1" os_prio=2 cpu=0.00ms elapsed=236.86s tid=0x00000156fc738800 nid=0x19f8 runnable  

"G1 Conc#2" os_prio=2 cpu=0.00ms elapsed=236.86s tid=0x00000156fc73e000 nid=0x9080 runnable  

"G1 Refine#0" os_prio=2 cpu=0.00ms elapsed=238.31s tid=0x00000156f92f1800 nid=0x9610 runnable  

"G1 Young RemSet Sampling" os_prio=2 cpu=0.00ms elapsed=238.31s tid=0x00000156deafe000 nid=0x81ec runnable  
"VM Periodic Task Thread" os_prio=2 cpu=0.00ms elapsed=238.18s tid=0x00000156f9f45000 nid=0x7dac waiting on condition  

JNI global refs: 369, weak refs: 86

`

**Comment by roiniti** [comment]
Ok, so a tiny workarround I think that can make this work (its a bit dirty), building a little localhost HTTP server and pass the uri to VLC so it manages all, (for what im doing its a good solution because it will simplify the implementation for vlc android that im also using), i don't know how this workarround sounds to you

**Comment by caprica** [comment]
How does it sound to me? Well, if it solves your problem, it sounds just fine to me. :-)

**Comment by roicbr** [comment]
Yeah but you could reproduce the problem? if libvlc stops reading the buffer at some point then all the CallbackMedia subclases are useless

**Comment by caprica** [comment]
If the problem is in libvlc, I can't/won't fix it.

If the problem is in vlcj, I will have to write a new sample and apparently wait 25 minutes each time to troubleshoot it. It may be the case in fact that the "normal" usage of the callback media is fine, and it's only cases like this custom streaming arrangement where there is a problem, and this report doesn't contain a sample.

So thanks for your comment, but time is precious. If I find some spare time, I will try, but I can't promise when.
